android/lib/src/main/java/com/marianhello/react/BackgroundGeolocationModule.java

```diff
@@ -433,8 +455,13 @@ private void sendEvent(ReactContext reactContext, String eventName, @Nullable Wr
     }
 
     public boolean hasPermissions() {
-        //TODO: implement
-        return true;
+        final Activity currentActivity = this.getCurrentActivity();
```

Comment:
actually @msde is right - you're passing a non-existent context variable to ContextCompat.checkSelfPermission instead of currentActivity